src/components/screens/meeting/new-meeting-screen.tsx

```diff
@@ -0,0 +1,252 @@
+import { MeetingsApiContext } from "../../providers/meetings-api-provider";
+import { ErrorContext } from "../../error-handler/error-handler";
+import MeetingNavigator from "../../../types/navigators/meeting";
+import { useNavigation, useRoute } from "@react-navigation/native";
+import { MeetingLanguage } from "../../../types";
+import { Button, Card, Divider, RadioButton, TextInput } from "react-native-paper";
+import strings from "../../../localization/strings";
+import { View } from "react-native";
+import { Text } from "react-native-paper";
+import { ScrollView } from "react-native-gesture-handler";
+import styles from "../../../styles/screens/meeting/new-meeting-screen";
+import theme from "../../../theme";
+import moment from "moment";
+
+/**
+ * New meeting screen
+ */
+const NewMeetingScreen: React.FC = () => {
+  const meetingsApiContext = React.useContext(MeetingsApiContext);
+  const navigation = useNavigation<MeetingNavigator.NavigationProps>();
+  const errorContext = React.useContext(ErrorContext);
+  const { params } = useRoute<MeetingNavigator.RouteProps>();
+  const meetingTime = params?.meetingTime;
+
+  const [ newMeeting, setNewMeeting ] = React.useState<Meeting>({
+    time: meetingTime?.startTime || new Date(),
+    contact: {
+      firstName: "",
+      lastName: ""
+    },
+    participantCount: 0,
+    type: MeetingType.Meeting,
+    language: MeetingLanguage.FI
+  });
+
+  if (!meetingTime) {
+    return null;
+  }
+
+  /**
+   * Validates new meeting 
+   */
+  const validateNewMeeting = () => {
+    return newMeeting.contact.firstName && newMeeting.contact.lastName && newMeeting.type && newMeeting.participantCount;
+  };
+
+  /**
+   * Handler for new meeting change 
+   *
+   * @param name name of the input
+   * @param value value of the input
+   */
+  const onNewMeetingChange = (name: string) => (value: string) => {
+    const updatedNewMeeting: Meeting = { ...newMeeting, [name]: value };
+    setNewMeeting(updatedNewMeeting);
+  };
+
+  /**
+   * Handler for meeting contact change 
+   *
+   * @param name name of the input
+   * @param value value of the input
+   */
+  const onNewMeetingContactChange = (name: string) => (value: string) => {
+    const updatedNewMeeting: Meeting = { ...newMeeting, contact: { ...newMeeting.contact, [name]: value } };
+    setNewMeeting(updatedNewMeeting);
+  };
+
+  /**
+   * Handler for meeting cancel 
+   */
+  const onMeetingCancel = () => {
+    navigation.navigate("meetingTimes");
+  };
+
+  /**
+   * Handler for meeting create 
+   */
+  const onMeetingCreate = async () => {
+    try {
+      await meetingsApiContext.createMeeting({ meeting: newMeeting });
+      setNewMeeting({
+        time: meetingTime?.startTime || new Date(),
+        contact: {
+          firstName: "",
+          lastName: ""
+        },
+        participantCount: 0,
+        type: MeetingType.Meeting,
+        language: MeetingLanguage.FI
+      });
+      navigation.navigate("meetingTimes");
```

Comment:
Could you use the created meeting from API instead of creating the meeting manually?

Reply:
This is setting the meeting back to initial meeting (clearing the selection) not setting the created meeting